test/specs/utils/deepMerge.spec.js

```diff
@@ -54,13 +54,13 @@ describe('utils::deepMerge', function () {
   });
 
   it('handles null and undefined arguments', function () {
-    expect(deepMerge(undefined, undefined)).toEqual({});
+    expect(deepMerge(undefined, undefined)).toEqual(undefined);
```

Comment:
Could you elaborate what is the worry?

deepMerge is an axios internal utility function so there should be no issue updating it as needed for axios functionality.

Reply:
No users will use `deepMerge` directly, but it may break axios internally. No changes is always better than changes, right? Unless we must change it with definite reasons. Suppose codes like,

```
config = deepMerge(a, b);
config.xx // possible to have undefined error
```

Reply:
It needs to be changed because the previous version loses information (it loses track of the fact that `undefined` was in the original object).

`deepMerge` is used in one place (in `mergeConfig`)  in the axios codebase: [see GitHub search results here](https://github.com/axios/axios/search?q=deepMerge&unscoped_q=deepMerge). Changing the function only affects that place.